testsuite/MDAnalysisTests/analysis/test_msd.py

```diff
@@ -83,133 +90,138 @@ def step_traj(NSTEP):  # constant velocity
-def test_fft_start_stop_step_all_dims(step_traj, NSTEP, dim, dim_factor):
-    # testing the fft algorithm on constant velocity trajectory
-    # test start stop step is working correctly
-    m_simple = MSD(step_traj, 'all', msd_type=dim, fft=True)
-    m_simple.run(start=10, stop=1000, step=10)
-    poly = characteristic_poly(NSTEP, dim_factor)
-    # polynomial must take offset start into account
-    assert_almost_equal(m_simple.timeseries, poly[0:990:10], decimal=3)
-
-
-def test_random_walk_u_simple(random_walk_u):
-    # regress against random_walk test data
-    msd_rw = MSD(random_walk_u, 'all', msd_type='xyz', fft=False)
-    msd_rw.run()
-    norm = np.linalg.norm(msd_rw.timeseries)
-    val = 3932.39927487146
-    assert_almost_equal(norm, val, decimal=5)
-
-
-def test_random_walk_u_fft(random_walk_u):
-    # regress against random_walk test data
-    msd_rw = MSD(random_walk_u, 'all', msd_type='xyz', fft=True)
-    msd_rw.run()
-    norm = np.linalg.norm(msd_rw.timeseries)
-    val = 3932.39927487146
-    assert_almost_equal(norm, val, decimal=5)
+class TestMSDSimple(object):
+
+    def test_selection_works(self, msd):
+        # test some basic size and shape things
+        assert_equal(msd.n_particles, 10)
+
+    @pytest.mark.parametrize("dim, dim_factor", [
+        ('xyz', 3), ('xy', 2), ('xz', 2), ('yz', 2), ('x', 1), ('y', 1),
+        ('z', 1)
+    ])
+    def test_simple_step_traj_all_dims(self, step_traj, NSTEP, dim,
+                                       dim_factor):
+        # testing the "simple" algorithm on constant velocity trajectory
+        # should fit the polynomial y=dim_factor*x**2
+        m_simple = MSD(step_traj, 'all', msd_type=dim, fft=False)
+        m_simple.run()
+        poly = characteristic_poly(NSTEP, dim_factor)
+        assert_almost_equal(m_simple.timeseries, poly, decimal=4)
+
+    @pytest.mark.parametrize("dim, dim_factor", [
+        ('xyz', 3), ('xy', 2), ('xz', 2), ('yz', 2), ('x', 1), ('y', 1),
+        ('z', 1)
+    ])
+    def test_simple_start_stop_step_all_dims(self, step_traj, NSTEP, dim,
+                                             dim_factor):
+        # testing the "simple" algorithm on constant velocity trajectory
+        # test start stop step is working correctly
+        m_simple = MSD(step_traj, 'all', msd_type=dim, fft=False)
+        m_simple.run(start=10, stop=1000, step=10)
+        poly = characteristic_poly(NSTEP, dim_factor)
+        # polynomial must take offset start into account
+        assert_almost_equal(m_simple.timeseries, poly[0:990:10], decimal=4)
+
+    def test_random_walk_u_simple(self, random_walk_u):
+        # regress against random_walk test data
+        msd_rw = MSD(random_walk_u, 'all', msd_type='xyz', fft=False)
+        msd_rw.run()
+        norm = np.linalg.norm(msd_rw.timeseries)
+        val = 3932.39927487146
+        assert_almost_equal(norm, val, decimal=5)
+
+
+def import_check(module):
```

Comment:
Unfortunately had to to do this to avoid doing pytest.importorskip everywhere (as far as I can tell you can just use pytest.importorskip at the class level, it just defaults to the whole module).

If we think it might be useful somewhere else, I can probably move this to MDAnalysisTests.utils though.

Reply:
I think maybe the way to do this is `pytest.mark.skipif()` then apply that to each class? This isn't hideous though

Reply:
Sorry, probably going to need clarification here, I'm not sure if you mean what's on line 157 or something different?

Reply:
Perhaps write the equivalent of `pytest.importorskip` that actually works at the class level, just like what you have, and put the new decorator to `utils.skipif_noimport` (or somesuch)... but what you have here is already pretty clear, especially if you put `import_check()` into utils.

Reply:
For simplicity I've just moved the function over to utils as `import_not_available`, it made more sense considering we already have a similar `executable_not_found` there.